src/main.cpp

```diff
@@ -897,7 +897,9 @@ bool ContextualCheckTransaction(const CTransaction& tx, CValidationState &state,
     
         // Check that all transactions are unexpired
         if (IsExpiredTx(tx, nHeight)) {
-            return state.DoS(dosLevel, error("ContextualCheckTransaction(): transaction is expired"), REJECT_INVALID, "tx-overwinter-expired");
+            // Don't increase banscore if the transaction only just expired
+            int expiredDosLevel = IsExpiredTx(tx, nHeight - 1) ? dosLevel : 0;
```

Comment:
Could an attacker with a large hash power store up more than one block and release them all at once, so that the race condition occurs over that many blocks rather than one?

Reply:
The attacker would need to be able to get the victim to propagate all those blocks within the race window, so the more blocks they need to propagate, the harder it becomes. But perhaps two is as feasible as one? We'd need to dig into when block propagation happens relative to block receiving.

Reply:
Yeah I was mainly thinking of two blocks as being the potential problem.

Reply:
(I'm such an incorrigible generalist, I literally find it hard to say the number two. -1, 0, 1 and n are the only important numbers.)